src/worker.c

```diff
@@ -119,43 +138,46 @@ static CURLMcode init(CURLM *cm, char *method, char *url, int timeout_millisecon
 
 	if (strcasecmp(method, "GET") == 0) {
 		if (reqBody) {
-			curl_easy_setopt(eh, CURLOPT_POSTFIELDS, reqBody);
-			curl_easy_setopt(eh, CURLOPT_CUSTOMREQUEST, "GET");
+			CURL_EZ_SETOPT(eh, CURLOPT_POSTFIELDS, reqBody);
+			CURL_EZ_SETOPT(eh, CURLOPT_CUSTOMREQUEST, "GET");
 		}
 	}
 
 	if (strcasecmp(method, "POST") == 0) {
 		if (reqBody) {
-			curl_easy_setopt(eh, CURLOPT_POSTFIELDS, reqBody);
+			CURL_EZ_SETOPT(eh, CURLOPT_POSTFIELDS, reqBody);
 		}
 		else {
-			curl_easy_setopt(eh, CURLOPT_POST, 1);
-			curl_easy_setopt(eh, CURLOPT_POSTFIELDSIZE, 0);
+			CURL_EZ_SETOPT(eh, CURLOPT_POST, 1);
+			CURL_EZ_SETOPT(eh, CURLOPT_POSTFIELDSIZE, 0);
 		}
 	}
 
 	if (strcasecmp(method, "DELETE") == 0) {
-		curl_easy_setopt(eh, CURLOPT_CUSTOMREQUEST, "DELETE");
+		CURL_EZ_SETOPT(eh, CURLOPT_CUSTOMREQUEST, "DELETE");
 	}
 
-	curl_easy_setopt(eh, CURLOPT_WRITEFUNCTION, body_cb);
-	curl_easy_setopt(eh, CURLOPT_WRITEDATA, cdata->body);
-	curl_easy_setopt(eh, CURLOPT_HEADERFUNCTION, header_cb);
-	curl_easy_setopt(eh, CURLOPT_HEADERDATA, cdata->response_headers);
-	curl_easy_setopt(eh, CURLOPT_HEADER, 0L);
-	curl_easy_setopt(eh, CURLOPT_URL, url);
-	curl_easy_setopt(eh, CURLOPT_HTTPHEADER, cdata->request_headers);
-	curl_easy_setopt(eh, CURLOPT_TIMEOUT_MS, timeout_milliseconds);
-	curl_easy_setopt(eh, CURLOPT_PRIVATE, cdata);
-	curl_easy_setopt(eh, CURLOPT_FOLLOWLOCATION, true);
+	CURL_EZ_SETOPT(eh, CURLOPT_WRITEFUNCTION, body_cb);
+	CURL_EZ_SETOPT(eh, CURLOPT_WRITEDATA, cdata->body);
+	CURL_EZ_SETOPT(eh, CURLOPT_HEADERFUNCTION, header_cb);
+	CURL_EZ_SETOPT(eh, CURLOPT_HEADERDATA, cdata->response_headers);
+	CURL_EZ_SETOPT(eh, CURLOPT_HEADER, 0L);
+	CURL_EZ_SETOPT(eh, CURLOPT_URL, url);
+	CURL_EZ_SETOPT(eh, CURLOPT_HTTPHEADER, cdata->request_headers);
```

Comment:
Assignment on line 137 doesn't affect this code because cdata->request_headers is set at line 135. Not sure how should this be but looks as 137 and 135 should be in reverse order. Also curl_slist_append at line 137 could output null at error, thus deleting the whole list instead of adding an element.

Reply:
Addressed both concerns. `curl_slist_append` is now tested for NULL.